addons/mrp/models/mrp_workorder.py

```diff
@@ -115,9 +115,9 @@ def _read_group_workcenter_id(self, workcenters, domain, order):
         'stock.move.line', 'workorder_id', 'Moves to Track',
         help="Inventory moves for which you must scan a lot number at this work order")
     finished_lot_id = fields.Many2one(
-        'stock.lot', string='Lot/Serial Number', compute='_compute_finished_lot_id',
-        inverse='_set_finished_lot_id', domain="[('product_id', '=', product_id), ('company_id', '=', company_id)]",
-        check_company=True)
+        'stock.lot', string='Lot/Serial Number', related='production_id.lot_producing_id',
```

Comment:
The line in the commit message "Also add the production end date to form and list view." seems not related to this commit